controller/Exporter.php

```diff
@@ -0,0 +1,169 @@
+                'success' => false,
+                'errorMsg' => $e->getMessage(),
+                'errorCode' => $e->getCode()
+            ]);
+        }
+    }
+
+    /**
+     * @param $parameters
+     * @return TaskInterface
+     * @throws \common_exception_Error
+     */
+    private function createExportTask(array $parameters)
+    {
+        $exportPackage = new ExportSynchronizationPackage();
+        $callable = $this->propagate($exportPackage);
+
+        /** @var QueueDispatcherInterface $queueService */
+        $queueService = $this->getServiceLocator()->get(QueueDispatcherInterface::SERVICE_ID);
+
+        $task = $queueService->createTask($callable, $parameters, self::TASK_LABEL);
+        $this->setLastExportTask($task);
+
+        return $task;
+    }
+
+    private function prepareTaskParameters()
+    {
+        $requestData = $this->getRequestParameters();
+        $this->validateRequestParameters($requestData);
+
+        return [
+            'organisation_id' => $requestData['organisation_id'],
+            'tao_version' => $this->getApplicationService()->getPlatformVersion(),
+            'box_id' => $this->getBoxId(),
+        ];
+    }
+
+    private function validateRequestParameters($data)
+    {
+        if (empty($data['organisation_id'])) {
+            throw new \common_exception_BadRequest('Parameter "organisation_id" is required.');
+        }
+    }
+
+    /**
+     * @return string
+     */
+    private function getBoxId()
+    {
+        return $this->getServiceLocator()->get(PublishingService::SERVICE_ID)->getBoxIdByAction(SynchronizeData::class);
+    }
+
+    /**
+     * @return ApplicationService
+     */
+    private function getApplicationService()
+    {
+        return $this->getServiceLocator()->get(ApplicationService::SERVICE_ID);
+    }
+
+    /**
+     * @throws \common_exception_RestApi
+     * @throws SyncExportException
+     */
+    private function checkExportPreconditions()
+    {
+        /** @var ExportService $exportService */
+        $exportService = $this->getServiceLocator()->get(ExportService::SERVICE_ID);
+        if (!$exportService->getOption(ExportService::OPTION_IS_ENABLED)) {
+            throw new SyncExportException(__('Synchronization export feature is not enabled.'));
+        }
+        $processedTaskStatusList = [
+            TaskLogInterface::STATUS_COMPLETED,
+            TaskLogInterface::STATUS_FAILED,
+            TaskLogInterface::STATUS_ARCHIVED,
+            TaskLogInterface::STATUS_CANCELLED
+        ];
+        $lastTask = $this->getLastExportTask();
+        if ($lastTask && !in_array($lastTask->getStatus(), $processedTaskStatusList)) {
+            throw new \common_exception_RestApi(__('The export is already running!'), 423);
+        }
+    }
+
+    /**
+     * @return EntityInterface|null
+     */
+    private function getLastExportTask()
+    {
+        try {
+            $taskId = $this->getCacheService()->get(self::EXPORT_TASK_CACHE_KEY);
```

Comment:
Could you please add organization identifier to the cache key?
In case when two sync managers from different organizations run this export first task will be overwritten.